core/embed/trezorhal/image.c

```diff
@@ -50,35 +50,71 @@ static secbool compute_pubkey(uint8_t sig_m, uint8_t sig_n,
   return sectrue * (0 == ed25519_cosi_combine_publickeys(res, keys, sig_m));
 }
 
-secbool load_image_header(const uint8_t *const data, const uint32_t magic,
-                          const uint32_t maxsize, uint8_t key_m, uint8_t key_n,
-                          const uint8_t *const *keys, image_header *const hdr) {
+secbool parse_image_header(const uint8_t *const data, const uint32_t magic,
+                           const uint32_t maxsize, image_header *const hdr) {
   memcpy(&hdr->magic, data, 4);
   if (hdr->magic != magic) return secfalse;
 
   memcpy(&hdr->hdrlen, data + 4, 4);
   if (hdr->hdrlen != IMAGE_HEADER_SIZE) return secfalse;
 
   memcpy(&hdr->expiry, data + 8, 4);
+  memcpy(&hdr->codelen, data + 12, 4);
+  memcpy(&hdr->version, data + 16, 4);
+  memcpy(&hdr->fix_version, data + 20, 4);
+  memcpy(&hdr->monotonic, data + 24, 1);
+  memcpy(&hdr->hw_model, data + 25, 1);
+  memcpy(&hdr->hw_revision, data + 26, 1);
+  memcpy(hdr->hashes, data + 32, 512);
```

Comment:
sigmask + signature are on a fixed offset, they start at `IMAGE_HEADER_SIZE - IMAGE_SIG_SIZE`

Reply:
okay tried to use the struct approach, but it kinda leads to somewhat larger diff. 

unfortunately, for vendor_header this is not possible since the header has variable amount of pubkeys and the struct needs fixed size, so we could only do this for part of the struct before the pubkeys, which doesn't seem worth it.

so the result doesn't look entirely elegant because of the two different approaches.

https://github.com/trezor/trezor-firmware/commit/df3eccea7ad47dae23ec6f4c1d67990f19a3e39b